src/BenchmarkDotNet.Core/Portability/RuntimeInformation.cs

```diff
@@ -321,5 +321,35 @@ internal static ICollection<Antivirus> GetAntivirusProducts()
             return Array.Empty<Antivirus>();
 #endif
         }
+
+
+        internal static VirtualMachineHypervisor GetVirtualMachineHypervisor()
+        {
+            VirtualMachineHypervisor[] hypervisors = { new HyperV(), new VirtualBox(), new VMware() };
```

Comment:
It would be better to make constructors of `HyperV`, `VirtualBox`, `VMware` private; introduce static instances in each class, and use them here.